datafusion/common/src/scalar.rs

```diff
@@ -3078,18 +3132,38 @@ impl fmt::Display for ScalarValue {
             ScalarValue::DurationMillisecond(e) => format_option!(f, e)?,
             ScalarValue::DurationMicrosecond(e) => format_option!(f, e)?,
             ScalarValue::DurationNanosecond(e) => format_option!(f, e)?,
-            ScalarValue::Struct(e, fields) => match e {
-                Some(l) => write!(
+            ScalarValue::Struct(struct_arr) => {
+                // ScalarValue Struct should always have a single element
+                assert_eq!(struct_arr.len(), 1);
+
+                let columns = struct_arr.columns();
+                let fields = struct_arr.fields();
+                let nulls = struct_arr.nulls();
+
+                write!(
```

Comment:
I think you can use the arrow display code here instead: https://docs.rs/arrow/latest/arrow/util/display/fn.array_value_to_string.html

Using ArrayFormatter directly would save a copy: https://docs.rs/arrow-cast/50.0.0/src/arrow_cast/display.rs.html#930